paasta_tools/cli/cmds/status.py

```diff
@@ -294,11 +300,20 @@ def paasta_status_on_api_endpoint(
     if status.git_sha != "":
         output.append("    Git sha:    %s (desired)" % status.git_sha)
 
-    instance_type = find_instance_type(status)
-
-    if system_paasta_config.get_enable_custom_cassandra_status_writer():
-        if status.get("cassandracluster") is not None:
-            instance_type = "cassandracluster"
+    if instance_type:
+        if status.get(instance_type) is None:
+            output.append(
+                PaastaColors.red(
+                    f"Status contains no data for {instance_type} instance type.\n"
+                )
+            )
+            return 1
+    else:
+        instance_type = find_instance_type(status)
+        if system_paasta_config.get_enable_custom_cassandra_status_writer():
```

Comment:
do you still need this flag if the user can now chose this when they run paasta status?

Reply:
That's correct, we won't need it anymore. I will remove it.